### ISSUE TITLE
No Sync'ed passwords?

### ISSUE THREAD

**Issue by bagage** [gnumdk/eolie#183]
TRANSLATORS:
https://translate.zanata.org/project/view/eolie

### Environment
- Eolie version: 0.9-27-g7b8cfb5
- GTK+ version: 3.22
- Operating system: Linux sessifet 4.9.0-1-amd64 #1 SMP Debian 4.9.6-3 (2017-01-28) x86_64 GNU/Linux

### Bug/Feature
I think Eolie support Firefox Sync for passwords since #39 is closed. But when I open `Privacy life` -> `Manage passwords`, only passwords that I did ask Eolie to save are displayed.
Similarly when I browse to a website with credentials in Firefox Sync, Eolie seems not to suggest it.

So my issue would be…:

1. Should Eolie display firefox Sync passwords in `manage passwords`? If not, can I see them somewhere?
2. Why Eolie is not filling forms where data should be present in Sync passwords?

**Comment by gnumdk** [comment]
>Should Eolie display firefox Sync passwords in manage passwords?

Yes

**Comment by gnumdk** [comment]
Just run eolie -d to get more informations on sync process.

**Comment by bagage** [comment]
So I created an account on Firefox and saved password. Then I started `eolie -d`, and it looks good:

```
local passwords: 1500484856.48, remote passwords: 1500570870.33
pull passwords
pulling {'id': '{ID}', 'modified': 1500570870.33, 'payload': {'timeCreated': 1500570726945, 'hostname': 'https://secure.anglaisfacile.com', 'passwordField': 'password', 'id': '{ID}', 'usernameField': 'email', 'username': 'EMAIL', 'formSubmitURL': 'https://secure.anglaisfacile.com', 'password': 'PASSWORD', 'timePasswordChanged': 1500570726945, 'httpRealm': None}}
local history: 1500570684.45, remote history: 1500570874.06
pull history
local bookmarks: 1500538489.57, remote bookmarks: 1500538489.57
push bookmarks
Stop syncing
pushing {'title': "!Apprendre l'anglais:Cours d'anglais,jeux,exercices,grammaire,tests-Enseigner l'anglais", 'visits': [{'date': 1500570763560000.0, 'type': 1}, {'date': 1500570649850000.0, 'type': 1}, {'date': 1500570822070000.0, 'type': 1}], 'id': 'SHz-tP8cZxVf', 'histUri': 'https://www.anglaisfacile.com'}
pushing {'title': 'Connectez-vous à notre club pour apprendre !', 'visits': [{'date': 1500570747060000.0, 'type': 1}, {'date': 1500570663100000.0, 'type': 1}, {'date': 1500570824220000.0, 'type': 1}], 'id': 'KGSaNToAddLX', 'histUri': 'https://secure.anglaisfacile.com/login.php'}
```

But once you open the website, Eolie does not suggest to use credentials. If you type them manually, it suggests you to save them though.

**Comment by gnumdk** [comment]
Is the password working in Firefox, because usernameField and passwordField used by YOUR Firefox are wrong. Maybe website updated "input" names... So if it works in Firefox, Firefox is using a fallback code...

**Comment by gnumdk** [comment]
Should not be working in your firefox too... Because I already have fallback code in Eolie but "Login input" is detected based on it name: email, login, username, ... 

On this site, the input name is "name" (who is coding this?). And I don't want to take a random input on page...

**Comment by bagage** [comment]
I can confirm you that it does work on Firefox, even with the bad entry names.
Same issue also happen on Amazon…

Actually don't you reproduce the issue? It happens for 100% of the websites for me.

**Comment by gnumdk** [comment]
I reproduce but only with your data, my firefox is putting goods value ...

**Comment by gnumdk** [comment]
Just tested with amazon:
- Close Eolie
- Launch Firefox, go to amazon, login and save password
-  Force sync in Firefox
- Launch Eolie, go to amazon, password is here...

There the question is: "Why your firefox is putting broken values?" What is Firefox version?

**Comment by gnumdk** [comment]
For amazon, launch seahorse and paste me information about amazon password, you should have:

formSubmitURL: https://www.amazon.com
userform: email
hostname: www.amazon.com
type: eolie web login
passform: password

**Comment by bagage** [comment]
OK, so I tested from a fresh account and tried login on various websites, most of them looked OK (inclu. amazon). I noted a few websites with issues, I'll report them separatly to keep this bug about one bug only.
After that, I removed all my Eolie data from my real account (using Eolie preferences "clean data" + `rm -r ~/.cache/eolie ~/.local/share/eolie`), and now most of websites are OK (except the ones which will have their bug report). Maybe it was some bad data from previous Eolie installation - gonna close it for now.
